extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/Tokens.java

```diff
@@ -46,26 +50,50 @@ public Long getRefreshTokenTimeSkew() {
     }
 
     public boolean isAccessTokenExpired() {
-        return isExpired(accessTokenExpiresAt);
+        final boolean expired = isExpired(accessTokenExpiresAt, true);
+        if (expired) {
+            LOG.debug("Access token has expired and must be refreshed");
+        }
+        return expired;
     }
 
     public boolean isRefreshTokenExpired() {
-        return isExpired(refreshTokenExpiresAt);
+        final boolean expired = isExpired(refreshTokenExpiresAt, false);
+        if (expired) {
+            LOG.debug("Refresh token has expired");
+        }
+        return expired;
     }
 
     public boolean isAccessTokenWithinRefreshInterval() {
         if (accessTokenExpiresAt == null || refreshTokenTimeSkew == null) {
             return false;
         }
         final long nowSecs = System.currentTimeMillis() / 1000;
-        return nowSecs + refreshTokenTimeSkew > accessTokenExpiresAt;
+        final boolean proactiveRefresh = nowSecs + refreshTokenTimeSkew > accessTokenExpiresAt;
+
+        if (proactiveRefresh) {
+            LOG.debugf(
+                    "Access token is still valid but must be refreshed because it will expire in about %d seconds"
+                            + " which is less than the refresh token time skew %d",
+                    accessTokenExpiresAt - nowSecs, refreshTokenTimeSkew);
+        }
+
+        return proactiveRefresh;
     }
 
-    private static boolean isExpired(Long expiresAt) {
+    private static boolean isExpired(Long expiresAt, boolean accessToken) {
         if (expiresAt == null) {
             return false;
         }
         final long nowSecs = System.currentTimeMillis() / 1000;
-        return nowSecs > expiresAt;
+        final boolean expired = nowSecs > expiresAt;
+
+        if (!expired) {
+            LOG.tracef((accessToken ? "Access " : "Refresh ") + " token is valid but will expire in about %d seconds",
+                    expiresAt - nowSecs);
```

Comment:
If using tracef, I wouldn't use string concatenation.

Reply:
@gsmet Hope it looks better now, thanks